### PR TITLE
CSP-483 - GET Admin

### PR DESCRIPTION
**Pull request by LiamWright**
/GET endpoint for Admin members

### CODE REVIEW

File: src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using SFA.DAS.AANHub.Domain.Entities;
+
+namespace SFA.DAS.AANHub.Application.Admins.Queries
+{
+    [ExcludeFromCodeCoverage]
+    public class GetAdminMemberResult
+    {
+        public Guid MemberId { get; set; }
+        public string? Email { get; set; }
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs:10]: Email and Name are both mandatory so why nullable here?

File: src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        public string? Name { get; set; }
+
+        public static implicit operator GetAdminMemberResult?(Admin? admin)
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs:13]: I do not like unnecessary null checks, is there a case where operator will be applied on null admin instance? otherwise please remove all the null checks and nullable fields.

File: src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMemberCommandValidator.cs [hunk 3]
```diff
@@ -7,25 +7,39 @@ namespace SFA.DAS.AANHub.Application.Employers.Commands
...
-                .NotEmpty();
-            RuleFor(c => c)
                 .NotEmpty()
-                .MustAsync(async (command, cancellation) =>
-                {
-                    var result = await employersReadRepository.GetEmployerByAccountIdAndUserId(command.AccountId, command.UserId);
-                    return result == null;
-                })
-                .WithMessage("UserId and AccountId pair already exist");
+                .MustAsync(async (command, _, _) => await CheckAccountIdAndUserIdPair(command.AccountId, command.UserId))
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMemberCommandValidator.cs:30]: Perhaps going forward unrelated impactful changes should be covered in a separate task/PR as QAs will miss these unless they are told.

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/AdminControllerTests.cs [hunk 4]
```diff
@@ -71,5 +73,105 @@ public AdminControllerTests()
...
+            var result = response as NotFoundResult;
+            result.Should().NotBeNull();
+            Assert.AreEqual(StatusCodes.Status404NotFound, result!.StatusCode);
+
+            mediatorMock.Verify(m => m.Send(It.IsAny<GetAdminMemberQuery>(), It.IsAny<CancellationToken>()));
+        }
+
+        [Test]
+        [AutoMoqData]
+        public async Task GetAdmin_InvokesQueryHandler_ResultGivesSuccessfulResult(
```
Comment (DevenKShah) [inline src/SFA.DAS.AANHub.Api.UnitTests/Controllers/AdminControllerTests.cs:123]: This is same as the first test, isn't it?

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberQueryValidator.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace SFA.DAS.AANHub.Application.Admins.Queries
+{
+    public class GetAdminMemberQueryValidator : AbstractValidator<GetAdminMemberQuery>
+    {
+        public GetAdminMemberQueryValidator()
+        {
+            RuleFor(a => a.UserName)
+                .NotNull()
+                .NotEmpty()
```
[the remark this answers is not in the document]
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberQueryValidator.cs]: Removed

File: src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberQueryValidator.cs [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+                .MaximumLength(200);
```
[the remark this answers is not in the document]
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberQueryValidator.cs]: Removed

File: src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs (hunk 1 above)
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs]: Nullable flag removed

File: src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs (hunk 2 above)
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Application/Admins/Queries/GetAdminMemberResult.cs:13]: I have removed the nullable flags on the properties, and within the operator. I have left the checks to see if the admin object is nullable to handle 404 situations

File: src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMemberCommandValidator.cs (hunk 3 above)
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMemberCommandValidator.cs:30]: Yes, myself Olaniyi agreed that the fix for CSP-556 would be part of this ticket to unblock the rest of the backlog, and that situations like this would not be regular thing going forward.

@OlaniyiOshinusi is aware of the additional test in this ticket.

File: src/SFA.DAS.AANHub.Api.UnitTests/Controllers/AdminControllerTests.cs (hunk 4 above)
Comment (LiamWright) [reply src/SFA.DAS.AANHub.Api.UnitTests/Controllers/AdminControllerTests.cs:123]: The first test is for the Create endpoint, this is for Get

File: src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMemberCommandValidator.cs (hunk 3 above)
Comment (DevenKShah) [reply src/SFA.DAS.AANHub.Application/Employers/Commands/CreateEmployerMemberCommandValidator.cs:30]: That's good then. Thanks
